params/precompile_config.go

```diff
@@ -209,6 +209,23 @@ func (c *ChainConfig) Get{YourPrecompile}Config(blockTimestamp *big.Int) *precom
 }
 */
 
+func (c *ChainConfig) GetActivePrecompileUpgrades(blockTimestamp *big.Int) PrecompileUpgrade {
+	pu := PrecompileUpgrade{}
+	if config := c.GetContractDeployerAllowListConfig(blockTimestamp); config != nil && !config.Disable {
+		pu.ContractDeployerAllowListConfig = config
+	}
+	if config := c.GetContractNativeMinterConfig(blockTimestamp); config != nil && !config.Disable {
+		pu.ContractNativeMinterConfig = config
+	}
+	if config := c.GetTxAllowListConfig(blockTimestamp); config != nil && !config.Disable {
+		pu.TxAllowListConfig = config
+	}
+	if config := c.GetFeeConfigManagerConfig(blockTimestamp); config != nil && !config.Disable {
+		pu.FeeManagerConfig = config
+	}
+	return pu
+}
```

Comment:
should we:
(a) add a comment here for user custom precompiles to add their precompile here
(b) switch to using the `precompileKeys` with a loop or calling one of the existing methods and checking disabled here (so the custom precompile user does not have to modify this method)
?

Reply:
I tried to use `precompileKeys` but we need the concrete type for marshalling. So in the end we have to convert `precompile.StatefulPrecompileConfig` to concrete types (same as `GetXXXConfig` functions). I added comments for now. I feel like we can remove these kind of manual iterations and most of the precompile duplications by expanding the precompile interface. I will give it a shot and create a different PR.

Reply:
That would be really nice to simplify this logic